src/test/scala/edu/ie3/simona/agent/participant/EvcsAgentModelCalculationSpec.scala

```diff
@@ -1987,6 +1992,425 @@ class EvcsAgentModelCalculationSpec
+        case ParticipantResultEvent(result: EvcsResult) =>
+          result.getInputModel shouldBe evcsInputModel.getUuid
+          result.getTime shouldBe 3600.toDateTime
+          result.getP should beEquivalentTo(33d.asKiloWatt)
+          result.getQ should beEquivalentTo(0d.asMegaVar)
+      }
+
+      // TICK 5400: ev2700 leaves
+
+      evService.send(
+        evcsAgent,
+        DepartingEvsRequest(5400, Seq(ev2700.uuid)),
+      )
+
+      evService.expectMsgType[DepartingEvsResponse] match {
+        case DepartingEvsResponse(evcs, evModels) =>
+          evcs shouldBe evcsInputModel.getUuid
+          evModels should have size 1
+          evModels.headOption match {
+            case Some(evModel) =>
+              evModel.uuid shouldBe ev2700.uuid
+              evModel.storedEnergy should approximate(KilowattHours(16.5))
+            case None => fail("Expected to get at least one ev.")
+          }
+      }
+
+      resultListener.receiveN(2).foreach {
+        case ParticipantResultEvent(result: EvResult) =>
+          result.getInputModel match {
+            case model if model == ev900.uuid || model == ev1800.uuid =>
+              throw new RuntimeException(
+                s"Ev $model already left but there are still results."
+              )
+
+            case model if model == ev2700.uuid =>
+              result.getTime match {
+                case time if time == 4500.toDateTime =>
+                  result.getP should beEquivalentTo(22d.asKiloWatt)
+                  result.getQ should beEquivalentTo(0d.asMegaVar)
+                  result.getSoc should beEquivalentTo(9.17.asPercent, 1e-2)
+                case time if time == 5400.toDateTime =>
+                  result.getP should beEquivalentTo(0d.asKiloWatt)
+                  result.getQ should beEquivalentTo(0d.asMegaVar)
+                  result.getSoc should beEquivalentTo(13.75.asPercent, 1e-2)
+                case _ =>
+                  throw new RuntimeException(
+                    s"Unexpected case at EvcsAgentModelCalculationSpec occurred."
+                  )
+              }
+
+            case _ =>
+              throw new RuntimeException(
+                s"Unexpected case at EvcsAgentModelCalculationSpec occurred."
+              )
+          }
+      }
+
+      resultListener.expectMsgType[ParticipantResultEvent] match {
+        case ParticipantResultEvent(result: EvcsResult) =>
+          result.getInputModel shouldBe evcsInputModel.getUuid
+          result.getTime shouldBe 4500.toDateTime
+          result.getP should beEquivalentTo(22d.asKiloWatt)
+          result.getQ should beEquivalentTo(0d.asMegaVar)
+      }
+      /* FixMe: We would expect another Evcs Result for the lastTick of 5400 here.
+         But this can't be calculated since there is no nextTick.
+         For simulation it is as well necessary to fix this e.g. by writing the lastResults when finishing simulation.
+       */
```

Comment:
This should be fixed with the new ParticipantAgent, but please just leave this in for now.